lib/team.rb

```diff
@@ -0,0 +1,54 @@
+require 'pry'
+require './lib/player.rb'
+
+class Team
+  attr_reader :country,
+              :players
+
+
+  def initialize(country)
+    @country = country
+    @players = []
+    @elimiation_status = false
+  end
+
+  #hmmmm i got stuck here,,, il come back later
+  def eliminated?()
+    @elimiation_status
+  end
+
+  def add_player(player)
+    @players << player
+  end
+
+
+# I am sure there is an easier way to do this  hahahahahah
+  def players_by_position(play_position)
+
+
+    @players.map do |player|
+      if player.position == play_position
+        wanted_players = []
+        wanted_players << player
+        return wanted_players
+
+      end
+    end
```

Comment:
Don't think this is quite doing what you need it to. `wanted_players` gets reset to an empty array each time you go through that block that you're passing to `map`. If you're going to use this pattern where you have some sort of empty thing that you fill as you move through an array, set it up *outside* of where you're iterating, and then also return it *outside* of where you're iterating.

```ruby
def players_by_position(play_position)
  wanted_players = []

  @players.each do |player|
    if player.position == play_position
      wanted_players << player
    end
  end

  return wanted_players
end
```